### DISCUSSION TITLE
开窗函数 LAST_VALUE的结果不正确

### DISCUSSION THREAD

**Discussion by zhudinghai** [apache/doris#15385] [RFC]
在Doris版本1.14上发现问题：
last_value(t1.job_status) over (partition by t1.staff_code order by t1.maint_date asc) last_job_status
执行的结果不正确

不过可以使用如下代替
first_value(t1.job_status) over (partition by t1.staff_code order by t1.maint_date desc) last_job_status

**Comment by gnehil** [comment]
建表语句方便发一下吗？

**Comment by morrySnow** [comment]
the frame of window function with order by is `RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW`, so the behavior meet expectation. If u want a `RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING` frame. u need to write an explicit frame.
